tests/foreman/cli/test_report.py

```diff
@@ -24,17 +24,17 @@
 from robottelo.cli.report import ConfigReport
 
 
-@pytest.fixture(scope='module', autouse=True)
-def run_puppet_agent(module_target_sat):
+@pytest.fixture(scope='module')
+def run_puppet_agent(session_puppet_enabled_sat):
```

Comment:
alright, so this will be an issue. This is running the action on a new satellite. However, your tests are interacting with the default satellite in the xdist session by using entities directly. You should try using the `session_puppet_enabled_sat` for entity interactions in the tests that rely on this fixture.